src/models.py

```diff
@@ -16,24 +24,69 @@
 
 logger = logging.getLogger(__name__)
 
+class MLP(nn.Module):
+    def __init__(self, num_hidden):
+        super().__init__()
+        self.num_hidden = num_hidden
+
+        self.lin1 = nn.Linear(num_hidden, num_hidden)
+        self.relu1 = nn.ReLU(inplace=True)
+       
+    def forward(self, x):
+        x = self.lin1(x)
+        x = self.relu1(x)
+        return x
+
+class ResNet(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.intermediate = {}  # Activation(s) of intermediate layers
+        # Pre-trainer Resnet34
+        self.pretrained = models.resnet34(pretrained=True)
+        # Attaching a hook to the last layer with informative feature vector
+        self.pretrained.layer4[2].bn2.register_forward_hook(self.forward_hook('BatchNorm'))
+        
+    def forward_hook(self,layer_name):
+        def hook(module, input, output):
+            self.intermediate.update({layer_name: output})
+        return hook
+
+    def forward(self, input):
+        out = self.pretrained(input)
+        return out, self.intermediate
```

Comment:
No need to return `self.intermediate`, since this is already an accesible property of the ResNet class.